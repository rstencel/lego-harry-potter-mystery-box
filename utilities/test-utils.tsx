import React, { FC, ReactElement } from "react"
import { render, RenderOptions } from "@testing-library/react"
import { IntlProvider } from "react-intl"
import messages from "../translations/en"

const AllTheProviders: FC<{ children: React.ReactNode }> = ({ children }) => (
  <IntlProvider locale="en" defaultLocale="en" messages={messages}>
    {children}
  </IntlProvider>
)

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options })

export { customRender as render }