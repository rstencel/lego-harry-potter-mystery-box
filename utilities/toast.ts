import { toast, TypeOptions } from 'react-toastify'

export const showToast = (message: string, type: TypeOptions, onCloseCallback?: () => void) => {
  toast.onChange(payload => {
    if (payload.status === "removed" && onCloseCallback) {
      onCloseCallback()
    }
  })
  
  toast(message, {
    type,
    position: "top-center",
    autoClose: 5000,
    hideProgressBar: false,
    closeOnClick: false,
    pauseOnHover: false,
    draggable: false,
    progress: undefined,
    theme: "light"
  })
}