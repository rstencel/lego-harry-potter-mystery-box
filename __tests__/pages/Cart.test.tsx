import Cart from "@/pages/cart/index"
import { render } from "../../utilities/test-utils"
import fetchMock from "jest-fetch-mock"

jest.mock('next/router', () => ({
  useRouter: () => ({
    push: jest.fn()
  })
}))

beforeEach(() => {
  fetchMock.doMock()
})

describe("Cart", () => {
  it("Should match snapshot", () => {
    const { container } = render(<Cart />)
    expect(container).toMatchSnapshot()
  })
})