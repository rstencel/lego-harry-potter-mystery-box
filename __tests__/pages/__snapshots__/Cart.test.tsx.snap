// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <main>
      <form
        class="container"
      >
        <div
          class="shippingDetails"
        >
          <h1>
            Shipping details
          </h1>
          <div
            class="form"
          >
            <div
              class="div1"
            >
              <label
                class="required-field"
                for="name"
              >
                Name
              </label>
              <input
                id="name"
                name="name"
                placeholder="Name"
              />
              <p
                class="error"
                style="visibility: hidden;"
              />
            </div>
            <div
              class="div2"
            >
              <label
                class="required-field"
                for="surname"
              >
                Surname
              </label>
              <input
                id="surname"
                name="surname"
                placeholder="Surname"
              />
              <p
                class="error"
                style="visibility: hidden;"
              />
            </div>
            <div
              class="div3"
            >
              <label
                for="phoneNumber"
              >
                Phone Number
              </label>
              <input
                id="phoneNumber"
                name="phoneNumber"
                placeholder="Phone Number"
                type="tel"
              />
              <p
                class="error"
                style="visibility: hidden;"
              />
            </div>
            <div
              class="div4"
            >
              <label
                class="required-field"
                for="email"
              >
                Email
              </label>
              <input
                id="email"
                name="email"
                placeholder="Email"
                type="tel"
              />
              <p
                class="error"
                style="visibility: hidden;"
              />
            </div>
            <div
              class="div5"
            >
              <label
                for="dateOfBirth"
              >
                Date of Birth
              </label>
              <input
                id="dateOfBirth"
                name="dateOfBirth"
                placeholder="Date of Birth"
                type="date"
              />
              <p
                class="error"
                style="visibility: hidden;"
              />
            </div>
            <div
              class="div6"
            >
              <label
                class="required-field"
                for="address"
              >
                Address
              </label>
              <input
                id="address"
                name="address"
                placeholder="Address"
              />
              <p
                class="error"
                style="visibility: hidden;"
              />
            </div>
            <div
              class="div7"
            >
              <label
                class="required-field"
                for="city"
              >
                City
              </label>
              <input
                id="city"
                name="city"
                placeholder="City"
              />
              <p
                class="error"
                style="visibility: hidden;"
              />
            </div>
            <div
              class="div8"
            >
              <label
                class="required-field"
                for="state"
              >
                State
              </label>
              <input
                id="state"
                name="state"
                placeholder="State"
              />
              <p
                class="error"
                style="visibility: hidden;"
              />
            </div>
            <div
              class="div9"
            >
              <label
                class="required-field"
                for="zipCode"
              >
                Zip Code
              </label>
              <input
                id="zipCode"
                name="zipCode"
                placeholder="Zip Code"
              />
              <p
                class="error"
                style="visibility: hidden;"
              />
            </div>
          </div>
        </div>
        <div
          class="card"
        >
          <h1
            style="color: black;"
          >
            Summary
          </h1>
          <div
            class="parts"
          >
            <p>
              There are  parts in this minifig:
            </p>
            <div
              class="loader-with-message"
              style="visibility: visible;"
            >
              <div
                class="loader"
              />
              <p
                style="color: black; margin-left: 1rem;"
              >
                Loading minifig parts ...
              </p>
            </div>
          </div>
          <button
            class="submit"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </main>
  </div>
</div>
`;
